.cbmc-batch/stubs/s_remove_node_override.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * FUNCTION: s_remove_node
+ *
+ * This function overrides the original implementation of the
+ * s_remove_node function from priority_queue.h with a no_op.
+ *
+ * It is safe to use as long as there are no assertions on the
+ * positions of elements in the priority queue after its invocation,
+ * as it doesn't really remove an element and reorders the rest, but
+ * it just reduces the length of both of the array lists of the
+ * priority queue by 1.
+ */
+
+#include <aws/common/priority_queue.h>
+
+int __CPROVER_file_local_priority_queue_c_s_remove_node(
+    struct aws_priority_queue *queue,
+    void *item,
+    size_t item_index) {
+    assert(aws_priority_queue_is_valid(queue));
+    assert(item && AWS_MEM_IS_WRITABLE(item, queue->container.item_size));
+    if (aws_array_list_get_at(&queue->container, item, item_index)) {
+        /* shouldn't happen, but if it does we've already raised an error... */
+        assert(aws_priority_queue_is_valid(queue));
+        return AWS_OP_ERR;
+    }
+
+    queue->container.length -= 1;
```

Comment:
Hmmm. We have only asserted that `queue` is valid. But it can be valid while having a length of zero, right? So this might underflow?

Please write a comment explaining why it's always okay to decrement `length`, it isn't obvious to me.

(I suppose the decrement below is safe because if `data` is non-null then length is at least 1, and that is checked by `aws_priority_queue_is_valid`?)

Reply:
This cannot underflow because `aws_array_list_get_at` would have failed (thus exiting early) if we tried to get an index that is not in the range of the container.

I will add a comment explaining the above.